snakemake/executors/google_lifesciences.py

```diff
@@ -582,10 +582,18 @@ def _set_snakefile(self):
         """
         from snakemake import SNAKEFILE_CHOICES
 
-        for snakefile in SNAKEFILE_CHOICES:
-            if os.path.exists(os.path.join(self.workdir, snakefile)):
-                self.snakefile = snakefile
-                break
+        # Case 1: snakefile specified on the command line
+        if os.path.exists(self.workflow.snakefile):
```

Comment:
One question here I'd want to know is if this could ever be None, or the attribute not defined at all? The default from `snakemake/__init__.py` seems to not allow this:
```python
    if args.snakefile is None:
        for p in SNAKEFILE_CHOICES:
            if os.path.exists(p):
                args.snakefile = p
                break
        if args.snakefile is None:
            print(
                "Error: no Snakefile found, tried {}.".format(
                    ", ".join(SNAKEFILE_CHOICES), file=sys.stderr
                )
            )
            sys.exit(1)
```
so I suspect it will be defined unless snakemake is called programatically (in python) and the snakefile explicitly set to None (does this happen?)

Reply:
Nope, that's not really allowed. Hence, you can remove the else.

Reply:
Removed then!